docs/source/howto/codes.rst

```diff
@@ -45,8 +45,92 @@ Design guidelines
 Running external codes
 ======================
 
-`#3987`_
+The process of running an external code through AiiDA involves loading and instantiating the appropriate ``CalcJob`` wrapping class for the code, filling the inputs necessary, and then running or submitting the constructed object.
 
+If the ``CalcJob`` has been registered in AiiDA (see section :ref:`on how to register plugins<how-to:plugins:entrypoints>`) you can use the ``CalculationFactory``:
+
+.. code-block:: python
+
+    from aiida.plugins import CalculationFactory
+    calculation_builder = CalculationFactory('arithmetic.add').get_builder()
+
+Otherwise, you can just load the class and get the builder directly:
+
+.. code-block:: python
+
+    from aiida.calculations.plugins.arithmetic.add import ArithmeticAddCalculation
+    calculation_builder = ArithmeticAddCalculation.get_builder()
+
+Then you need to start filling out the builder with all the inputs that you need to specify to run your calculation.
+In the case of the ``ArithmeticAddCalculation``, there are 4 required inputs:
+
+.. code-block:: python
+    :linenos:
+
+    # Specific for ArithmeticAdd
+    spec.input('x', valid_type=(orm.Int, orm.Float), help='The left operand.')
+    spec.input('y', valid_type=(orm.Int, orm.Float), help='The right operand.')
+
+    # General for all CalcJobs
+    spec.input('code', valid_type=orm.Code, help='The `Code` to use for this job.')
+    spec.input('metadata.options.resources', valid_type=dict, required=True, validator=validate_resources,
+        help='Set the dictionary of resources to be used by the scheduler plugin, like the number of nodes, '
+             'cpus etc. This dictionary is scheduler-plugin dependent. Look at the documentation of the '
+             'scheduler for more details.')
+
+The first three are input nodes (lines 2, 3 and 6), so you can either use nodes that are already in your database (you can query for them, load them, etc.) or you can create new nodes and these will be automatically stored when starting the calculation.
+
+.. code-block:: python
+
+    code_node = orm.load_node(100)
+    num1_node = orm.Int(17)
+    num2_node = orm.Int(11)
+
+    calculation_builder.code = code_node
+    calculation_builder.x = num1_node
+    calculation_builder.y = num2_node
+
+The last input (line 7) is not a node but a dictionary containing information about the resources required from the remote machine to run the code.
```

Comment:
This line is incorrect I think. You are not adding the metadata here, but in the next block

Reply:
I guess it is no longer relevant if I'm removing the metadata, but the line "The last input (line 7)" refers not to the last code block, but the previous one. I was hoping it would have been a bit more clear looking at the compiled doc because it was the only code block with numbers in the section, but it is true that it is not ideally phrased.